### PR TITLE
Setup Deployment

### PR DESCRIPTION
**Pull request by rosalinekarr**
Problem
=======
We want to automatically deploy to GitHub Pages
[#176962224](https://www.pivotaltracker.com/story/show/176962224)

Solution
========
Added GitHub Actions to auto-deploy to `gh-pages` branch

with @enddynayn

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    branches:
+      - main
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+
+      - name: Install and Build
```
Comment (wilwade) [inline .github/workflows/main.yml:13]: Do we want / need to cache? https://github.com/actions/cache/blob/main/examples.md#node---npm

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Build and Deploy
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
```
Comment (wilwade) [inline .github/workflows/main.yml:8]: Can we specify which node version to use?

File: package.json [hunk 3]
```diff
@@ -2,22 +2,27 @@
...
-        "@emotion/react": "11.1.5",
-        "@emotion/styled": "11.1.5",
-        "framer-motion": "3.3.0",
-        "dokz": "2.0.5",
-        "next": "10.0.7",
-        "react": "17.0.1",
-        "react-dom": "17.0.1"
+        "@emotion/react": "^11.1.1",
+        "@emotion/styled": "^11.0.0",
+        "docz": "latest",
```
Comment (wilwade) [inline package.json:20]: We might want to specify the version here to limit package-lock churn.

File: package.json [hunk 4]
```diff
@@ -2,22 +2,27 @@
...
+        "react": "^16.8.6",
```
Comment (wilwade) [inline package.json:21]: We don't get 17? :(

File: doczrc.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+export default {
+  base: "/spec/",
+  typescript: true,
+  ignore: ["CODEOFCONDUCT.md", "LICENSE.md", "README.md"],
+  menu: [
```
Comment (wilwade) [inline doczrc.js:5]: I recall that the other setup auto added to this. Do new files get auto added if running in dev mode? If not do we need to add this to the readme as a note to add stuff?

**Review by wilwade — CHANGES_REQUESTED** [review]
Few notes, but looking good.

File: .github/workflows/main.yml (hunk 1 above)
Comment (rosalinekarr) [reply .github/workflows/main.yml:13]: We probably don't need to, but I can go ahead and add it. Currently, our workflow runs in just under half a second.

File: doczrc.js (hunk 5 above)
Comment (rosalinekarr) [reply doczrc.js:5]: New files will automatically be added, but if you want to control where they appear in the sidebar, we'll have to add them here. Dokz actually did the same thing, but it buried this config in `pages/.app.jsx`.

File: package.json [hunk 6]
```diff
@@ -2,22 +2,27 @@
...
-        "react-dom": "17.0.1"
+        "@emotion/react": "^11.1.1",
+        "@emotion/styled": "^11.0.0",
+        "docz": "^2.3.1",
+        "react": "^17.0.1",
+        "react-dom": "^16.8.6"
     },
     "devDependencies": {
-        "compose-function": "3.0.3"
+        "@types/react": "^16.8.23",
```
Comment (wilwade) [inline package.json:25]: React / Type version mismatch.

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+name: Build and Deploy
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-16.04
```
Comment (wilwade) [inline .github/workflows/main.yml:8]: Isn't 16.04 EOL'd?

File: package.json [hunk 8]
```diff
@@ -2,22 +2,27 @@
...
-        "framer-motion": "3.3.0",
-        "dokz": "2.0.5",
-        "next": "10.0.7",
-        "react": "17.0.1",
-        "react-dom": "17.0.1"
+        "@emotion/react": "^11.1.1",
+        "@emotion/styled": "^11.0.0",
+        "docz": "^2.3.1",
+        "react": "^17.0.1",
+        "react-dom": "^16.8.6"
```
Comment (wilwade) [inline package.json:22]: Should this match the react version?

**Review by wilwade — DISMISSED** [review]
Three more questions.

File: package.json (hunk 6 above)
Comment (rosalinekarr) [reply package.json]: fixed

File: .github/workflows/main.yml (hunk 7 above)
Comment (rosalinekarr) [reply .github/workflows/main.yml]: changed to 20.04

File: package.json (hunk 8 above)
Comment (rosalinekarr) [reply package.json]: fixed

**Comment by enddynayn** [comment]
👍

**Pull request merged**